### PR TITLE
finish step undefined status

### PR DESCRIPTION
**Pull request by dgosantos89**
I had the same problem that was reported at Issue [58](https://github.com/reportportal/agent-js-postman/issues/58) and Issue [64](https://github.com/reportportal/agent-js-postman/issues/64)

Some steps was marked with In progress, because the step was undefined, so my solution is finish all steps that was undefined status as SKIPPED status

### CODE REVIEW

**Comment by orgads** [comment]
@AmsterGet ping?

**Comment by AmsterGet** [comment]
Hi guys!
I'm going to review the PR this week.

**Comment by dgosantos89** [comment]
Hi @AmsterGet!
Any news about this fix?

**Comment by brunotanoue** [comment]
Hi @AmsterGet. How are you?

I'm really interested to use reportportal in my company with Postman and Azure Pipelines, but the issues [58](https://github.com/reportportal/agent-js-postman/issues/58) and [64](https://github.com/reportportal/agent-js-postman/issues/64) are resulting in undefined status in some cases. Is there any deadline?

File: lib/reporter.js [hunk 1]
```diff
@@ -441,7 +456,7 @@ class Reporter {
     finishSuite () {
         const currentSuiteTempId = this.getCurrentSuiteTempId();
         const status = this.collectionRunOptions.collection &&
-            utils.getCollectionVariablesByKey('status', this.collectionRunOptions.collection.variables);
+            utils.getCollectionVariablesByKey('status', this.collectionRunOptions.collection.variables) || 'PASSED';
```
Comment (AmsterGet) [inline lib/reporter.js:459]: The ReportPortal API will calculate the final suite status based on the statuses of its descendants, so setting here `PASSED` as a fallback may lead to wrong suite status in the system.

File: lib/reporter.js [hunk 2]
```diff
@@ -325,12 +325,27 @@ class Reporter {
             .promise.catch(errorHandler);
     }
 
+    finishUndefinedStep() {
+        for (const item of this.collectionMap.values()) {
+            for (const step of item.steps) {
+                if (step.status === undefined) {
+                    let finishTestItemRQ = {
+                        endTime: new Date().valueOf(),
+                        status: 'SKIPPED'
```
Comment (AmsterGet) [inline lib/reporter.js:334]: Please use status as constant defined at `TestStatus`

**Review by AmsterGet — CHANGES_REQUESTED** [review]

**Comment by AmsterGet** [comment]
Hi guys!
Sorry, there have been very busy weeks and unexpected leaves in the team.

Thanks @dgosantos89 for the solution. Please check my comments above.

@brunotanoue Good to hear that.
We plan to release a fix once it's merged, so it might take some time.
Also, @Bam6ycha is currently working on a backquoted test names issue fix (https://github.com/reportportal/agent-js-postman/issues/60) and we plan to release those fixes together.

We will keep you updated in this topic.

File: lib/reporter.js (hunk 2 above)
Comment (dgosantos89) [reply lib/reporter.js]: Oh, thanks! I just fixed this

**Comment by Bam6ycha** [comment]
Hi,@dgosantos89. I hope you are doing well.
This issue was fixed by the already merged PR.
We really appreciate you for your help and time.

This issue occurred because we didn't support conditional tests.
That is why we decided to implement this feature in the next releases.

**Comment by Bam6ycha** [comment]
Closed due to irrelevance

**Comment by Bam6ycha** [comment]
@dgosantos89, FYI: The new version of the agent has been released. You can test it out with your tests.

**Pull request closed without merging**